server/server.go

```diff
@@ -127,113 +120,91 @@ func (s *Server) HandleEvent(ctx context.Context, e lookout.Event) error {
 	return err
 }
 
-// HandleReview sends request to analyzers concurrently
-func (s *Server) HandleReview(ctx context.Context, e *lookout.ReviewEvent, safePosting bool) error {
-	ctx, logger := ctxlog.WithLogFields(ctx, log.Fields{
-		"provider": e.Provider,
-	})
-	logger.Infof("processing pull request")
+func (s *Server) AnalyzeAndComment(ctx context.Context, e lookout.Event, safePosting bool) error {
+	comments, err := s.analyze(ctx, e, safePosting)
+	if err == nil {
+		if err := s.post(ctx, e, comments, safePosting); err != nil {
+			s.status(ctx, e, lookout.ErrorAnalysisStatus)
+			err = fmt.Errorf("posting analysis failed: %s", err)
+		} else {
+			s.status(ctx, e, lookout.SuccessAnalysisStatus)
+		}
+	}
+	return err
+}
 
+func (s *Server) analyze(ctx context.Context, e lookout.Event, safePosting bool) ([]lookout.AnalyzerComments, error) {
+	ctxlog.Get(ctx).Infof("processing event type %d", e.Type())
+
+	var comments []lookout.AnalyzerComments
 	if err := e.Validate(); err != nil {
-		return err
+		return comments, err
```

Comment:
we use to avoid indirections when doing an early return caused by an error; example:

```go
func doSomething() (string, error) {
    var msg string
    // do things, operate with msg...
    // at  some point, we check an error, and we do:
    if err := checkSomething(); err != nil {
        return "", err // instead of `return msg, err`; relying on a previous value of `msg`
    }
}
```

That way, we do explicit the things that we return